File: crates/zyntax_lua/pressure/large_strings.lua
-- A string past the pool's largest class built every step and left in
-- a global, where only the collector reclaims the one before: each is
-- a block of its own, outside the pool.
local base = string.rep("x", 4096)
held = ""

local function main(n)
  local total = 0
  for i = 1, n do
    held = base .. i
    total = total + #held
  end
  return total
end

print(main(tonumber(arg[1])))

File: crates/zyntax_python/pressure/large_strings.py
# A string past the pool's largest class built every step and stored
# into a field, where the last one is left for the collector: each is
# a block of its own, outside the pool. The accumulator is an int.
import sys

class Holder:
    def __init__(self, s: str):
        self.s = s

def main(n: int) -> int:
    total = 0
    base = "x" * 4096
    h = Holder("")
    for i in range(n):
        h.s = base + str(i)
        total += len(h.s)
    return total

print(main(int(sys.argv[1])))
